wire/error.go

```diff
@@ -17,19 +141,46 @@ import (
 // differentiate between general io errors such as io.EOF and issues that
 // resulted from malformed messages.
 type MessageError struct {
-	Func        string // Function name
-	Description string // Human readable description of the issue
+	ErrorCode   ErrorCode // Describes the kind of error
+	Description string    // Human readable description of the issue
 }
 
 // Error satisfies the error interface and prints human-readable errors.
-func (e *MessageError) Error() string {
-	if e.Func != "" {
-		return fmt.Sprintf("%v: %v", e.Func, e.Description)
-	}
-	return e.Description
+func (m MessageError) Error() string {
```

Comment:
In the same nature as the other comment, I'd also implement `Unwrap()` by making it return the underlying `ErrorCode` error.

It might be useful to also implement `Is()` and then verify whether the passed target is an `ErrorCode` so you can avoid an extra Unwrap call for the most common use case.